group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -480,9 +482,35 @@ public CompletableFuture<List<OffsetFetchResponseData.OffsetFetchResponseTopics>
             return FutureUtils.failedFuture(Errors.COORDINATOR_NOT_AVAILABLE.exception());
         }
 
-        return FutureUtils.failedFuture(Errors.UNSUPPORTED_VERSION.exception(
-            "This API is not implemented yet."
-        ));
+        // For backwards compatibility, we support fetch commits for the empty groupId.
```

Comment:
When we say we support fetch commits for the empty group ID, does this mean that we support it by always returning an error?

Reply:
No. It means that you can actually commit offsets to a group names with an empty group id. I was not aware of this until I rewrote this code.

Reply:
Ok -- I guess I just wasn't sure if this comment and the error need to be changed or not. Is the idea you can commit them but can't fetch them?

Reply:
Talked to David offline -- concluded that the comment was a little confusing -- null groupIds are not allowed but empty ones are.